opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt

```diff
@@ -84,7 +86,10 @@ internal object ObservabilityIndex {
      */
     @Suppress("TooGenericExceptionCaught")
     private fun createIndex() {
-        if (!isIndexExists(INDEX_NAME)) {
+        if (isIndexExists(INDEX_NAME)) {
+            // TODO: Only update mappings when they have changed so it doesn't run on every request
+            updateMappings()
```

Comment:
Should this change happen when users use the updated observability plugin for the first time. Similar to what we have for old notebooks index. You can check when you look at this in detail.

Reply:
You're right, I'll take a look at how the old notebooks index does it and add it in a future PR! 😄